### DISCUSSION TITLE
why the Interfaces property of INamedTypeSymbol，return all the interfaces result witch is same as AllInterfaces property

### DISCUSSION THREAD

**Discussion by MarvelTiter** [dotnet/roslyn#75619] [Q&A]
the code maybe like this，IWrapHello interface's `INamedTypeSymbol` 
```CSharp
public interface IHello
{
}

public interface IHello<T>: IHello
{
}

public interface IWrapHello: IHello<int>
{
}
```
```Csharp
// IWrapHello 
INamedTypeSymbol symbol ;
// symbol.Interfaces == symbol.AllInterfaces  why?
```

**Comment by 333fred** [comment]
Interface inheritance does not work like class inheritance. Interfaces directly implement all interfaces that were inherited from any base interfaces.

**Reply by MarvelTiter** [reply]
> Interface inheritance does not work like class inheritance. Interfaces directly implement all interfaces that were inherited from any base interfaces.

 if I need to recursively handle interface inheritance, are there other way to achieve this?

**Reply by CyrusNajmabadi** [reply]
Why do you need to do that?
